vm/systemSmartContracts/delegation.go

```diff
@@ -1023,6 +1038,7 @@ func (d *delegation) withdraw(args *vmcommon.ContractCallInput) vmcommon.ReturnC
 				break
 			}
 		}
+		delegator.UnStakedFunds = append(delegator.UnStakedFunds[:fundIndex], delegator.UnStakedFunds[fundIndex+1:]...)
```

Comment:
this is not safe. good find. However take the unbonded fundIndex into a list and add a delete function after.